modules/mqtt/destination/mqtt-worker.c

```diff
@@ -183,67 +182,24 @@ _insert(LogThreadedDestWorker *s, LogMessage *msg)
   */
 }
 
-static gint
-_log_ssl_errors(const gchar *str, gsize len, gpointer u)
-{
-  MQTTDestinationWorker *self = (MQTTDestinationWorker *) u;
-
-  msg_error("MQTT TLS error", evt_tag_printf("line", "%.*s", (gint) len, str),
-            log_pipe_location_tag(&self->super.owner->super.super.super));
-  return TRUE;
-}
-
-static MQTTClient_SSLOptions
-_create_ssl_options(MQTTDestinationWorker *self)
-{
-  MQTTDestinationDriver *owner = (MQTTDestinationDriver *) self->super.owner;
-
-  MQTTClient_SSLOptions ssl_opts = MQTTClient_SSLOptions_initializer;
-  ssl_opts.trustStore = owner->ca_file;
-  ssl_opts.CApath = owner->ca_dir;
-  ssl_opts.keyStore = owner->cert_file;
-  ssl_opts.privateKey = owner->key_file;
-  ssl_opts.enabledCipherSuites = owner->ciphers;
-  ssl_opts.sslVersion = owner->ssl_version;
-  ssl_opts.enableServerCertAuth = owner->peer_verify;
-  ssl_opts.verify = owner->peer_verify;
-  ssl_opts.disableDefaultTrustStore = !owner->use_system_cert_store;
-  ssl_opts.ssl_error_cb = _log_ssl_errors;
-  ssl_opts.ssl_error_context = self;
-
-  return ssl_opts;
-}
-
 static gboolean
 _connect(LogThreadedDestWorker *s)
 {
   MQTTDestinationWorker *self = (MQTTDestinationWorker *)s;
   MQTTDestinationDriver *owner = (MQTTDestinationDriver *) s->owner;
-  gint rc;
-
-  MQTTClient_connectOptions conn_opts = MQTTClient_connectOptions_initializer;
-  conn_opts.keepAliveInterval = owner->keepalive;
-  conn_opts.cleansession = FALSE;
-  conn_opts.username = owner->username;
-  conn_opts.password = owner->password;
 
-#if SYSLOG_NG_HAVE_PAHO_HTTP_PROXY
-  if (owner->http_proxy)
-    {
-      conn_opts.httpProxy = owner->http_proxy;
-      conn_opts.httpsProxy = owner->http_proxy;
-    }
-#endif
+  gint rc;
 
-  MQTTClient_SSLOptions ssl_opts = _create_ssl_options(self);
-  conn_opts.ssl = &ssl_opts;
+  MQTTClient_connectOptions conn_opts;
+  MQTTClient_SSLOptions ssl_opts;
+  mqtt_client_options_to_mqtt_client_connection_option(&owner->options, &conn_opts, &ssl_opts);
```

Comment:
Can you rename this to something like `mqtt_client_options_to_MQTTClient_connectOptions`?

IMO, it would be better to see the exact output type in the function name.